mitosheet/src/utils/formatColumns.tsx

```diff
@@ -1,346 +0,0 @@
-import React from "react"
-import MitoAPI from "../jupyter/api"
-import DropdownItem from "../components/elements/DropdownItem"
-import { getSelectedNumberSeriesColumnIDs } from "../components/endo/selectionUtils"
-import { ColumnID, FormatType, FormatTypeObj, GridState, MitoSelection, SheetData } from "../types"
-import DropdownCheckmark from '../components/icons/DropdownCheckmark'
-import { isNumberDtype } from "./dtypes"
-
-export const FORMAT_DISABLED_MESSAGE = 'You must have at least one Number column selected to adjust the formatting.'
-
-
-const formatCellDataAsStringWithCommas = (cellData: string | number | boolean, decimalPlaces?: number): string => {
-    return Number(cellData).toLocaleString("en-US", {minimumFractionDigits: decimalPlaces, maximumFractionDigits: decimalPlaces})
-}
-
-/*
-    For a cell to be formatted as a number, the cell must only contain valid number symbols
-    and be either a int or float column. 
-
-    Note: The cellData type of a float is a string.
-*/
-export const displayCellAsNumber = (cellData: boolean | string | number, columnDtype: string): boolean => {
-    // If the column is not a number series, then don't format the cell as a number
-    if (!isNumberDtype(columnDtype)) {
-        return false
-    }
-
-    if (typeof cellData === 'boolean') {
-        return false
-    } else if (typeof cellData === 'number') {
-        return true
-    } else {
-        return isStringNumeric(cellData)
-    }
-}
-
-/*
-    Determines if a string is a number
-    Adapted from: https://stackoverflow.com/questions/175739/how-can-i-check-if-a-string-is-a-valid-number
-    Note: This won't format numbers that have non . separators correctly, but that's okay for now because we wouldn't want to switch them to decimals
-*/
-export const isStringNumeric = (str: string): boolean => {
-    if (typeof str != "string") return false // we only process strings!  
-    return !isNaN(str as unknown as number) && // use type coercion to parse the _entirety_ of the string (`parseFloat` alone does not do this)...
-           !isNaN(parseFloat(str)) // ...and ensure strings of whitespace fail
-}
-
-/*
-    Returns cellData formatted as a number with decimals if the cell only contains valid number symbols
-    and the columnMitoType is a number_series. Otherwise, returns the unaltered cellData
-*/
-export const formatCellData = (cellData: boolean | string | number, columnDtype: string, columnFormatType: FormatTypeObj): string => {
-    // Wrap in a try, catch because there are lots of type cases and we'd rather be safe than sorry.
-    try {
```

Comment:
One thing I want to call your attention to in this function: there is a _ton_ of rightward drive. The most indented thing in inside a try, inside a if, inside a switch, inside a case, inside an if, inside an if. 

This is really really hard to maintain code. Every line you edit, you have to think about 5 different conditions that must be true. It's also practically just hard to move things about, and simplify cases, as things are so nested.

Check out the new code in format.tsx. Instead of checking the positive versions of things, it checks the opposite of the conditions, and returns early. This allows us to reduce a level or two of the rightward drift. 

Also, because we simplify the datatypes (as mentioned above), we can really just make the most simple checks you could imagine. 

Please take 5 minutes or so to look at the `formatCellData` here, and the new `formatCellData`, and let me know what you think!

Reply:
Yeah that makes a lot of sense. I think the fact that the code had is a great sign that checking the opposite condition first would've simplified things

```
        } else {
            return '' + cellData
        }
    } catch {
        return '' + cellData 
    }

```